crates/bevy_state/src/app.rs

```diff
@@ -57,8 +57,16 @@ pub trait AppExtStates {
     fn enable_state_scoped_entities<S: States>(&mut self) -> &mut Self;
 }
 
+/// Separate function to only warn once for all state installation methods.
+fn ensure_states_plugin_is_installed(app: &SubApp) {
```

Comment:
Nit: Technically this doesn't _ensure_ that the plugin is installed, since the plugin will still not be installed after calling this function. I'd give this a more accurate name like `warn_if_no_states_plugin`.

Reply:
fair